src/project/types/website/listing/website-listing-read.ts

```diff
@@ -679,14 +687,32 @@ function validateItem(
   }
 }
 
-function listItemFromMeta(meta: Metadata) {
-  const listingItem = cloneDeep(meta);
+async function listItemFromMeta(
+  meta: Metadata,
+  project: ProjectContext,
+  listing: ListingDehydrated,
+) {
+  let listingItem = cloneDeep(meta);
 
   // If there is a path, try to complete the filename and
   // modified values
-  if (meta.path !== undefined) {
-    meta[kFieldFileName] = basename(meta.path as string);
-    meta[kFieldFileModified] = fileModifiedDate(meta.path as string);
+  if (typeof meta.path === "string") {
+    const base = basename(meta.path);
+    const extension = extname(base).toLocaleLowerCase();
+    meta[kFieldFileName] = base;
+    meta[kFieldFileModified] = fileModifiedDate(meta.path);
+
+    const markdownExtensions = [".qmd", ".md", ".rmd"];
+    if (markdownExtensions.indexOf(extension) !== -1) {
+      const fileListing = await listItemFromFile(meta.path, project, listing);
```

Comment:
This function will return `undefined` for files that are explicitly marked as `draft` (we used this to exclude draft files from appearing in listings). I think perhaps we should respect that here since the concept is that the object notation is just a way to inject additional props... What do you think?

Reply:
Do you mean that if `listItemFromFile` returns undefined we should return undefined instead of simply not populating the template? That's a good idea.

The reason to not do it, on the other hand, is that we might have custom listing code that assumes entries are objects (and now they can be object || undefined).